### PR TITLE
Bundle Items price after saving to 0.00 go back to default price value Bundle Product Edit.

### LINKED ISSUE #32383: Bundle Items price after saving to 0.00 go back to default price value Bundle Product Edit
### Preconditions (*)
1. Magento 2.4.2
1. Magento 2.4-develop

### Steps to reproduce (*)
1. Make a Bundle Product on the admin dashboard
1. Set the Bundle Items price to zero (0.00)
1. Save it
1. Go back again to the Bundle Items price you will see the price is not zero it will display the child product price.

### Expected result (*)
1. Saving it with Bundle Items zero price when you go back to the Bundle product edit it should be zero price. 
image

### Actual result (*)
1. Save the Bundle Product with bundle items price zero.
image

1. When you go back to the Bundle Product Edit the bundle item price displayed the item default price.
image

### Testing notes

The case from https://github.com/magento/magento2/issues/28431 needs to be tested as well during verification of this issue fix

## 
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- < ] Severity: **S0** *- Affects critical data or functionality and leaves users without workaround.*
- [x] Severity: **S1** *- Affects critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S2** *- Affects non-critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S3** *- Affects non-critical data or functionality and does not force users to employ a workaround.*
- [ > Severity: **S4** *- Affects aesthetics, professional look and feel, “quality” or “usability”.*

### PR DESCRIPTION
**Pull request by engcom-Kilo**
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/32383

### Manual testing scenarios (*)

1. see https://github.com/magento/magento2/issues/32383

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by zakdma** [comment]
@magento run Functional Tests EE, Unit Tests

File: dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+use Magento\Catalog\Model\Product\Type;
+use Magento\Catalog\Model\Product\Visibility;
+use Magento\TestFramework\Bundle\Model\PrepareBundleLinks;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\Workaround\Override\Fixture\Resolver;
+
+Resolver::getInstance()->requireDataFixture('Magento/Bundle/_files/multiple_products.php');
+
+$objectManager = Bootstrap::getObjectManager();
+$productRepository = $objectManager->create(ProductRepositoryInterface::class);
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php]: please use ->get() for ProductRepositoryInterface

File: dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    ->setSku('bundle_product')
+    ->setVisibility(Visibility::VISIBILITY_BOTH)
+    ->setStatus(Status::STATUS_ENABLED)
+    ->setStockData(['use_config_manage_stock' => 1, 'qty' => 100, 'is_qty_decimal' => 0, 'is_in_stock' => 1])
+    ->setPriceView(0)
+    ->setPriceType(Price::PRICE_TYPE_FIXED)
+    ->setPrice(110.0)
+    ->setShipmentType(0);
+
+$OptionData = [
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php]: Variables names should be $camelCase

File: dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+use Magento\TestFramework\Workaround\Override\Fixture\Resolver;
+
+Resolver::getInstance()->requireDataFixture('Magento/Bundle/_files/multiple_products.php');
+
+$objectManager = Bootstrap::getObjectManager();
+$productRepository = $objectManager->create(ProductRepositoryInterface::class);
+$prepareBundleLinks = $objectManager->get(PrepareBundleLinks::class);
+$product = $objectManager->create(Product::class);
+$product->setTypeId(Type::TYPE_BUNDLE)
+    ->setAttributeSetId(4)
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php]: Use $product->getDefaultAttributeSet() instead

File: dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection_rollback.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Registry;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\Workaround\Override\Fixture\Resolver;
+
+Resolver::getInstance()->requireDataFixture('Magento/Bundle/_files/multiple_products_rollback.php');
+$objectManager = Bootstrap::getObjectManager();
+$registry = $objectManager->get(Registry::class);
+$productRepository = $objectManager->create(ProductRepositoryInterface::class);
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection_rollback.php]: please use ->get() for ProductRepositoryInterface

**Review by zakdma — CHANGES_REQUESTED** [review]

File: dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php (hunk 3 above)
Comment (engcom-Kilo) [reply dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php]: Done.

File: dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php (hunk 2 above)
Comment (engcom-Kilo) [reply dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php]: Fixed

File: dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php (hunk 1 above)
Comment (engcom-Kilo) [reply dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection.php]: Done

File: dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection_rollback.php (hunk 4 above)
Comment (engcom-Kilo) [reply dev/tests/integration/testsuite/Magento/Bundle/_files/fixed_bundle_product_zero_price_option_selection_rollback.php]: Done

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Review by zakdma — APPROVED** [review]

**Comment by dmitriyprime** [comment]
:heavy_check_mark:  QA passed

**Before:**
before

**After:**
after

**Comment by dmitriyprime** [comment]
The case from https://github.com/magento/magento2/issues/28431 is not reproduced.

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-32594`.

**Pull request merged**